### DISCUSSION TITLE
Can or should we include some sort of RHDS usage tracking?

### DISCUSSION THREAD

**Discussion by markcaron** [RedHat-UX/red-hat-design-system#339] [Ideas]
Can or should we include some sort of RHDS usage tracking?

I don't have a specific solution as to how this could be accomplished, but I believe that having some sort of _opt-in_ tracking of RHDS usage across websites could help us with our KPIs.

How many sites are using RHDS? How many components are being used, and which components?

Open to ideas as to how to accomplish this, as well as concerns about performance hits. We definitely don't want to slow down performance, so how can we do this with the smallest footprint?

**Comment by zeroedin** [comment]
I think the best way to do this would be if we were loading components from a CDN.  Then you offload the tracking bits to the CDN itself but alas we don't have that capability... yet.  Asking a component to phone home might be too much overhead. 

Said this in chat expanding on the "phoning home":

> Another option is that we could build a registry component on the page that collected all <rh-*> tags and it would report back to a backend to store the URL + components on the page.  Would be asking more of FED's to include the registry component on every page, but should be easy with drupal templating to add something like that, and we'd have to build that registry backend.  I still like the CDN way though as the ultimate goal.

**Reply by bennypowers** [reply]
hard agree. IMO the only 'tracking' code we should include would be to catch specific or uncaught errors from components and send them to an alerts mechanism. "who's loading stuff" can be handled well by our CDN. Page loading perf can be measured independently, instead of doing it inside the components
